lld/MachO/ObjC.cpp

```diff
@@ -1270,7 +1274,9 @@ void ObjcCategoryMerger::eraseMergedCategories() {
 
       eraseISec(catInfo.catBodyIsec);
 
-      tryEraseDefinedAtIsecOffset(catInfo.catBodyIsec, catLayout.nameOffset);
+      // We can't erase 'catLayout.nameOffset' because for Swift categories, the
+      // name will be referenced in __METACLASS_DATA_.
+      // TODO: handle the above smarter
```

Comment:
I'm not sure I follow. Is this the same behavior with ld64?
What is __METACLASS_DATA__? -- do you have a reference to it?
What's the consequence if we leave it without deleting it?

Reply:
I will address this in a follow-up PR with a test that includes `__METACLASS_DATA_`.